ivy/functional/frontends/numpy/random/functions.py

```diff
@@ -39,3 +41,39 @@ def geometric(p, size=None):
 @from_zero_dim_arrays_to_scalar
 def normal(loc=0.0, scale=1.0, size=None):
     return ivy.random_normal(mean=loc, std=scale, shape=size, dtype="float64")
+
+
+@to_ivy_arrays_and_back
+@from_zero_dim_arrays_to_scalar
+def permutation(x):
+    if isinstance(x, int):
+        new_array = ivy.arange(x)
+        return new_array
+    elif isinstance(x, list) or isinstance(x, tuple):
+        x_len = float(len(x))
+        position_list = []
+        arr_list = []
+        while len(position_list) < x_len:
+            r_position = round(rd.uniform(1.0, x_len))
+            if r_position not in position_list:
+                position_list.append(r_position)
+                arr_list.append(x[r_position - 1])
+        new_array = ivy.array(arr_list)
+        return new_array
+    else:
+        try:
+            arr_shape = ivy.shape(x, as_array=False)
+            first_dim = float(arr_shape[0])
+            position_list = []
+            arr_list = []
+            while len(position_list) < first_dim:
+                r_position = round(rd.uniform(1.0, first_dim))
+                if r_position not in position_list:
+                    position_list.append(r_position)
+                    arr_list.append(x[r_position - 1])
+
+            new_array = ivy.array(arr_list).reshape(arr_shape)
+            return new_array
```

Comment:
Probably all of this can be replaced with [`ivy.shuffle`](https://lets-unify.ai/ivy/functional/ivy/random.html#ivy.functional.ivy.random.shuffle)